app/models/user.rb

```diff
@@ -1,2 +1,34 @@
 class User < ActiveRecord::Base
+=begin
+User attributes
+username:   this is the username that the user goes by it can be used to log in. 
+       It showed be shown next to users posts and comments. S
+       Should be unique
+password:   personal password used to log in 
+email:     user's email addres it can be used to log in
+=end
+  has_many :posts
+  has_many :comments
+
+  EMAIL_REGEX =  %r{[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}\z}i
+
+  validates :username, presence: true, uniqueness: true, length: { in: 3..20 }
+  validates :email, presence: true, uniqueness: true, format: EMAIL_REGEX 
+  validates :password, confirmation: true, length: {in: 6..20}  
```

Comment:
Nitpick, can we make the password max at least 32? Preferably something like 256?